src/_P003_Pulse.ino

```diff
@@ -277,14 +279,14 @@ void Plugin_003_pulsecheck(byte Index)
   // time. Very rare.
   //  Alternatively there is timePassedSince(Plugin_003_pulseTimePrevious[Index]); but this is not in IRAM at this time, so do not use in a
   // ISR!
-  const unsigned long PulseTime = millis() - Plugin_003_pulseTimePrevious[Index];
+  const unsigned long PulseTime = micros() - Plugin_003_pulseTimePrevious[Index];
```

Comment:
This will cause strange issues when the `micros()` will overflow.
Was already an issue with `millis()` but that would only happen every 49.7 days.
`micros()` will overflow roughly 0.0497 days :) (71 minutes)
So that will happen quite often.

You should use the `timePassedSince` call for this.
Or maybe use `micros64()` and store `Plugin_003_pulseTimePrevious` and `Plugin_003_pulseTime` should be stored as `uint64_t`

Reply:
Hmm.. I am running this for over 2 days now on a node with 2 pulse input's at roughly 2000RPM and didn't have any issue so far. I guess what happens is, that you will just "loose" that one event when the counter overflows.... which is not nice, but not really a problem I think...
the bigger problem is, that you do not want to do too much in this routine as it's time critical...
I'm happy to look into it, but as said, I don't thik that it's a real problem...

Reply:
At high pulse rates it is not likely to be a problem, but the pulse plugin is also used to measure pulses which may be more than an hour apart.
For example water consumption counter.

Reply:
latest commit uses `uint64_t micros64()` internally